packages/common/client-utils/src/events/emitter.ts

```diff
@@ -3,68 +3,45 @@
  * Licensed under the MIT License.
  */
 
-import { UsageError } from "@fluidframework/telemetry-utils/internal";
-import { getOrCreate } from "../util/index.js";
-import type { Listenable, Listeners, Off } from "./listeners.js";
+import type {
+	Listenable,
+	Listeners,
+	Off,
+	HasListeners,
+	NoListenersCallback,
+	IEmitter,
+} from "@fluidframework/core-interfaces/internal";
 
 /**
- * Interface for an event emitter that can emit typed events to subscribed listeners.
+ * Subset of Map interface.
+ * @internal
  */
-export interface IEmitter<TListeners extends Listeners<TListeners>> {
-	/**
-	 * Emits an event with the specified name and arguments, notifying all subscribers by calling their registered listener functions.
-	 * @param eventName - the name of the event to fire
-	 * @param args - the arguments passed to the event listener functions
-	 */
-	emit<K extends keyof Listeners<TListeners>>(
-		eventName: K,
-		...args: Parameters<TListeners[K]>
-	): void;
-
-	/**
-	 * Emits an event with the specified name and arguments, notifying all subscribers by calling their registered listener functions.
-	 * It also collects the return values of all listeners into an array.
-	 *
-	 * Warning: This method should be used with caution. It deviates from the standard event-based integration pattern as creates substantial coupling between the emitter and its listeners.
-	 * For the majority of use-cases it is recommended to use the standard {@link IEmitter.emit} functionality.
-	 * @param eventName - the name of the event to fire
-	 * @param args - the arguments passed to the event listener functions
-	 * @returns An array of the return values of each listener, preserving the order listeners were called.
-	 */
-	emitAndCollect<K extends keyof Listeners<TListeners>>(
-		eventName: K,
-		...args: Parameters<TListeners[K]>
-	): ReturnType<TListeners[K]>[];
+interface MapGetSet<K, V> {
+	get(key: K): V | undefined;
+	set(key: K, value: V): void;
```

Comment:
Good question. I'm not sure how conscious of a decision it was originally, but having the return type of `set` be looser means it's a wee bit easier to have an object satisfy `MapGetSet`. I think the Tree package has some objects which are _not_ JS maps, but which satisfy `MapGetSet`, and can therefore be used with e.g. `getOrCreate`. At best it'd be annoying to have to change their `set` method return `this` in order to satisfy the interface, and at worst it would be prohibitive because the return value is already being used for something else. I think it's a good call considering that I rarely (have I ever?) seen use of the fluent-style code that would leverage the returned `this` - e.g. `map.set(...).size` or something like that - so nothing is really lost.

You made me wonder if we could still make it more closely derived:

```typescript
interface MapGetSet<K, V> extends Pick<Map<K, V>, "get"> {
    set(...args: Parameters<Map<K,V>["set"]>): void
}
```

The "typescript enjoyer and purist" part of me thinks that is cool and good, whereas the rational part of me thinks that it is less readable and will bother subsequent readers more than help them :)